test/Select.spec.tsx

```diff
@@ -31,47 +27,62 @@ describe('Select', () => {
     );
   }
 
-  function isListBoxRendered(node: Node) {
-    return node instanceof HTMLElement && node.localName.includes('list-box');
-  }
-
-  async function assert(container: HTMLElement) {
-    const select = container.querySelector('vaadin-select');
-    expect(select).to.exist;
-
-    const valueButton = select!.querySelector('vaadin-select-value-button');
-    expect(valueButton).to.exist;
-
-    valueButton!.dispatchEvent(new PointerEvent('click', { bubbles: true }));
+  async function assert(user: ReturnType<UserEvent['setup']>) {
+    const select = await findByQuerySelector('vaadin-select');
 
-    const overlay = document.querySelector('vaadin-select-overlay');
-    expect(overlay).to.exist;
+    const valueButton = await findByQuerySelector('vaadin-select-value-button', select);
+    expect(valueButton).to.have.text('Bar');
 
-    await catchRender(overlay!, isListBoxRendered);
+    await user.click(valueButton);
 
+    const overlay = await findByQuerySelector('vaadin-select-overlay');
     expect(overlay).to.have.text('FooBar');
   }
 
-  afterEach(cleanup);
-  afterEach(catcher);
+  let user: ReturnType<UserEvent['setup']>;
+
+  beforeEach(() => {
+    user = userEvent.setup();
+  });
 
   it('should use children if no renderer property set', async () => {
-    const { container } = render(<Select ref={ref} items={items} value="bar" />);
-    await assert(container);
+    render(<Select items={items} value="bar" />);
+    await assert(user);
   });
 
   it('should use renderer prop if it is set', async () => {
-    const { container } = render(<Select ref={ref} items={items} renderer={Renderer} value="bar" />);
-    await assert(container);
+    render(<Select renderer={Renderer} value="bar" />);
+    await assert(user);
   });
 
   it('should use children render function as a renderer prop', async () => {
```

Comment:
I think it would be good to also have a test verifying that the value is set correctly when the `renderer` function is passed through `children`.